drivers/audio/dmic_nrfx_pdm.c

```diff
@@ -132,6 +133,37 @@ static bool check_pdm_frequencies(const struct dmic_nrfx_pdm_drv_cfg *drv_cfg,
 	uint32_t req_rate = pdm_cfg->streams[0].pcm_rate;
 	bool better_found = false;
 
+#if NRF_PDM_HAS_PRESCALER
+	uint32_t src_freq = 32 * 1000 * 1000UL;
+	uint32_t req_freq = req_rate * ratio;
+	uint32_t prescaler = src_freq / req_freq;
+	uint32_t act_freq = src_freq / prescaler;
+
+	if (act_freq >= pdm_cfg->io.min_pdm_clk_freq && act_freq <= pdm_cfg->io.max_pdm_clk_freq &&
+		is_better(act_freq, ratio, req_rate, best_diff, best_rate, best_freq)) {
```

Comment:
Such check occurs three times in this file. I think it would be good to use a helper function for it, to improve code readability.